backend/courses/management/commands/compute_review_semester_department.py

```diff
@@ -0,0 +1,77 @@
+import json
+import re
+from textwrap import dedent
+
+from click import BaseCommand
+from django.db.models import OuterRef
+from tqdm import tqdm
+
+from alert.management.commands.recomputestats import get_semesters
+from courses.models import Department
+from review.annotations import review_averages
+
+
+def average_by_dept(fields, path=None):
+    """
+    For each department and year, compute the average of given fields
+    (see `alert.models.ReviewBit` for an enumeration of fields) across all (valid) sections.
+    Note that if fields should be a list of strings representing the review fields to be aggregated.
+    """
+    dept_avgs = {}
+    for semester in tqdm(get_semesters(semesters="all")):
+        semester_dept_avgs = review_averages(
+            Department.objects.all(),
+            fields=fields,
+            subfilters={
+                "review__section__course__semester": semester,
+                "review__section__course__department__id": OuterRef("id"),
```

Comment:
Small detail but you can access the `id` field with `department_id` rather than `department__id` to avoid a join.

Reply:
Interesting. I'll make the change